### ISSUE TITLE
Annotations are not evaluated correctly when using a variable for the datasource (and repeat rows)

### ISSUE THREAD

**Issue by jmdacruz** [grafana/grafana#39539]
**What happened**: I have a dashboard with a variable (`$datasource`) for selecting multiple Prometheus datasources (corresponding to different environments), and I'm using this variable to "repeat" a row for each of the datasources. When I add an annotation for this dashboard (e.g., deployment events, alerts), using `$datasource` as the datasource for the annotation, I would expect each of the repeated rows to have annotations evaluated against the datasource they are repeated for, but instead it looks like the first datasource on the list of selected items for the `$datasource` variable is used to generate the annotations for every row. It looks like annotations are evaluated only once for the entire dashboard, which in this case is not correct. Haven't tested this with "repeat panels" but I assume the same would happen.

**What you expected to happen**: If rows or panels are repeated for a variable that represents a datasource, then annotations that use this same variable should be evaluated in the right context.

**How to reproduce it (as minimally and precisely as possible)**: Create a dashboard that has a variable for selecting multiple Prometheus data sources, and add a row set to "repeat" on the datasource. Then add an annotation that uses this same variable for the datasource, and have the annotation should Prometheus alerts. 

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.1.5 (and also tested with 6.6.2)
- Data source type & version: Prometheus 2.19.2
- OS Grafana is installed on: Ubuntu (reproduced locally with docker image grafana/grafana:8.1.5-ubuntu)
- User OS & Browser: Chrome/Safari/Firefox
- Grafana plugins: None
- Others:

**Comment by jmdacruz** [comment]
This may be a duplicate of #4953 (and maybe #717)

**Comment by torkelo** [comment]
annotation queries are global and not executed per panel, an annotation query that uses a muli value data source variable as the data source should not work (it is not supported), or it might just take the first value if it works

**Comment by torkelo** [comment]
for this to work we would first need to implement https://github.com/grafana/grafana/issues/717

**Comment by akapelrud** [comment]
Now that https://github.com/grafana/grafana/issues/717 is implemented, maybe it's time to reopen and fix this issue? The lack of proper variable-specific annotation for repeated panels is a showstopper.

This would be an extremely valuable feature to have, allowing using dashboard variables to select between e.g. different sensors — which could be numbered in the hundreds to thousands, applicable to research facilities, laboratories and/or industrial monitoring systems.

**Comment by morsik** [comment]
Same issue here on Grafana 12.1.1.

@torkelo Since #717 is now done, this should be totally doable now. Can this be reopened? ;)

My case: Kubernetes dashboard with volume panels repeated on `$pvc` variable. I have a Loki annotation query for PVC resize events (filtering by `pvc="$pvc"`) - but the annotation shows up on *all* repeated panels, not just the one for the PVC that was actually resized.

`filter.ids` (Selected panels) doesn't help either - it targets the template panel ID, but repeated instances get dynamic IDs so you can't scope per-instance.

Would be great to have annotation queries respect scoped variables in repeated panels.
